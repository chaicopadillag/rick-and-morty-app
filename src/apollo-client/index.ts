import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { getEnvironments } from "../env";
console.log(getEnvironments().VITE_BASE_URL);
export const apolloClient = new ApolloClient({
  link: new HttpLink({
    uri: process.env.VITE_BASE_URL || "http://localhost:4000/graphql",
  }),
  cache: new InMemoryCache(),
});
