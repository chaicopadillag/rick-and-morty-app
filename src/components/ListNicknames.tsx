import { FC } from "react";

type ListProp = {
  names: string[];
  order: "ASC" | "DESC";
};

/**
 *
 * @param param {ListProp}
 * @returns
 */
export const ListNicknames: FC<ListProp> = ({ names, order }) => {
  names.sort((a, b) =>
    order === "ASC"
      ? a.trim().localeCompare(b.trim(), "es", { sensitivity: "base" })
      : b.trim().localeCompare(a.trim(), "es", { sensitivity: "base" })
  );

  return (
    <ul>
      {names.map((name, index) => (
        <li key={index}>{name}</li>
      ))}
    </ul>
  );
};
