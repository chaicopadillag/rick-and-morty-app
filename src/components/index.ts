export * from './layouts/MainLayout';
export * from './character/CharacterCard';
