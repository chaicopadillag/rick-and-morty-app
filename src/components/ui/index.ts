export * from "./Navbar";
export * from "./CustomCard";
