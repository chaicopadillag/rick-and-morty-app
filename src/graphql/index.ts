import { gql } from '@apollo/client';

export const CHARACTERS_QUERY = gql`
  query {
    characters {
      results {
        id
        name
        image
        species
      }
      info {
        count
        pages
        next
        prev
      }
    }
  }
`;

export const CHARACTER_QUERY = gql`
  query findCharacter($id: ID!) {
    character(id: $id) {
      id
      name
      species
      image
    }
  }
`;
