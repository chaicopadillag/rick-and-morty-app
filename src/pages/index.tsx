export * from "./HomePage";
export * from "./CharacterPage";
export * from "./PageNotFound";
export * from "./FavoritesPage";
