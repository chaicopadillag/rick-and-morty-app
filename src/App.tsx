import { Routes, Route, Link } from 'react-router-dom';
import { CharacterPage, HomePage } from './pages';

const App = () => {
  return (
    <>
      <Routes>
        <Route path='/' element={<HomePage />} />
        <Route path='/character/:characterId' element={<CharacterPage />} />
        <Route path='/favorites' element={<Link to='/favorites'>Favoritos</Link>} />
      </Routes>
    </>
  );
};

export default App;
