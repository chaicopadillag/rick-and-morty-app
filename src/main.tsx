import React from "react";
import ReactDOM from "react-dom/client";
import { NextUIProvider } from "@nextui-org/react";

import App from "./App";
import "./css/index.css";
import { darkTheme } from "./themes";
import { Provider } from "react-redux";
import { store } from "./store";

ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
    <NextUIProvider theme={darkTheme}>
      <Provider store={store}>
        <App />
      </Provider>
    </NextUIProvider>
  </React.StrictMode>
);
