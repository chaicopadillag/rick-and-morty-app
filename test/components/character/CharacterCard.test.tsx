import React from 'react';
import { render } from '@testing-library/react';
import { CharacterCard } from '../../../src/components';
import { characterDemo } from '../../fixture/characterFixture';
import { MemoryRouter } from 'react-router-dom';

describe('Prueba Snapshot de Componente <CharacterCard/>', () => {
  test('debe de mostrar el componente', () => {
    const { container } = render(
      <MemoryRouter>
        <CharacterCard character={characterDemo} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
