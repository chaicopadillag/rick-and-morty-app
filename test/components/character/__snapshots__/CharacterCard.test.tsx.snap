// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Snapshot de Componente <CharacterCard/> debe de mostrar el componente 1`] = `
<div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-iiUTsGP-css nextui-grid-item xs sm md xl"
  >
    <div
      class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-PJLV nextui-c-BDLTQ-kUADSW-variant-bordered nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true nextui-c-PJLV-gulvcB-isFocusVisible-false"
      role="button"
      tabindex="0"
    >
      <div
        class="nextui-c-eFfoBo nextui-c-eFfoBo-ifRCjYa-css"
      >
        <div
          class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-idqeSIq-css nextui-image-container nextui-image--loading"
          data-state="loading"
        >
          <div
            class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
          />
          <img
            alt="Rick Sanchez"
            class="nextui-c-kbhVdb nextui-c-kbhVdb-iUwpmy-css nextui-image"
            data-state="loading"
            height="200"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            width="100%"
          />
        </div>
      </div>
      <div
        class="nextui-c-cakKrd"
      >
        <div
          class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-idqFNMl-css"
        >
          <b
            class="nextui-c-PJLV nextui-c-PJLV-iIzlms-css"
          >
            Rick Sanchez
          </b>
        </div>
      </div>
    </div>
  </div>
</div>
`;
